jwt_proxy/api.py

```diff
@@ -67,13 +96,16 @@ def validate_jwt(relative_path):
         )
     except jwt.exceptions.ExpiredSignatureError:
         return jsonify(message="token expired"), 401
-
-    response_content = proxy_request(
-        req=request,
-        upstream_url=f"{current_app.config['UPSTREAM_SERVER']}/{relative_path}",
-        user_info=decoded_token.get("email") or decoded_token.get("preferred_username"),
-    )
-    return response_content
+    
+    # TODO: call new function here to dynamically load a filter call dependent on config; hardwired for now
+    if scope_filter(request, decoded_token):
+        response_content = proxy_request(
+            req=request,
+            upstream_url=f"{current_app.config['UPSTREAM_SERVER']}/{relative_path}",
+            user_info=decoded_token.get("email") or decoded_token.get("preferred_username"),
+        )
+        return response_content
+    return jsonify(message="invalid request"), 400
```

Comment:
as this implies a lack of authorization, a 401 is a more accurate code.

Reply:
Agreed that 400 generally doesn't fit. I'd advocate to keep the 400 response on parsing errors like [get_json failing](https://tedboy.github.io/flask/_modules/flask/wrappers.html#Request.on_json_loading_failed) though.
Would a 403 in response be an even [better semantic fit](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403), as by this point the user is authenticated but unauthorized to make the request, and reauthenticating will not help?

Reply:
Disregard that first bit, I didn't realize werkzeug's BadRequest errors automatically return 400 when left uncaught. They sure do think these things through 😛